knative-operator/pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -184,6 +183,12 @@ func (r *ReconcileKnativeServing) Reconcile(request reconcile.Request) (reconcil
 			return reconcile.Result{}, fmt.Errorf("failed to update status: %w", err)
 		}
 	}
+
+	if instance.Status.IsReady() {
+		common.KnativeServingReadyG.Set(1)
+	} else {
+		common.KnativeServingReadyG.Set(0)
+	}
```

Comment:
This will break if a second `KnativeServing` instance gets installed in a cluster, right? Like if the user creates it in the wrong namespace as we've had users do in the past?

Also, does `reconcileErr` need to be taken into account in case we had a reconcilation error? Do we want to unconditionally set the metric value even if reconciling failed?

Tagging @jcrossley3 in because he may know about the multiple `KnativeServing` issue, at least.

Reply:
Currently, we have a validating webhook that rejects any create/update request that's not in a specific namespace (configured via env var), and there can only be one in that namespace.

In the future, when knative compoonents will all be installed in a single `openshift-serverless` namespace, we need slightly different behavior: https://github.com/openshift-knative/serverless-operator/pull/386/commits/196765214b6646da3b76132a35c39679a335c3bb

But the bottom line is that we should always be using a validating webhook of some sort to ensure that only one instance is installed per cluster. HTH

Reply:
My assumption was that metrics have namespace as an attribute (at least they show this way in prometheus when I checked them) and since we could have openshift-serverless-1, openshift-serverless-2 why this would break?